src/generator/from_proto.rs

```diff
@@ -23,6 +23,36 @@ struct Context {
     query: String,
 }
 
+/// A custom enum representing either a valid value or an error.
+#[derive(Debug)]
+enum Valid<T> {
```

Comment:
actually valid is already defined in `valid.rs` we need to use that instead of defining a new one

Reply:
Sorry for the delay, I have used the Valid in valid.rs